pkg/githubutil/releases.go

```diff
@@ -13,14 +13,51 @@ import (
 	"github.com/upfluence/ubuild/pkg/version"
 )
 
-func GetLastVersion(repo string) (*version.Version, error) {
-	splittedRepo := strings.Split(repo, "/")
-	r, _, err := buildClient().Repositories.GetLatestRelease(
-		context.Background(),
-		splittedRepo[len(splittedRepo)-2],
-		splittedRepo[len(splittedRepo)-1],
+const (
+	maxPage = 3
+	perPage = 10
+)
+
+func GetLatestRelease(ctx context.Context, repo string) (*github.RepositoryRelease, error) {
+	var (
+		release *github.RepositoryRelease
+
+		client       = buildClient().Repositories
+		page         = 1
+		splittedRepo = strings.Split(repo, "/")
+		owner        = splittedRepo[len(splittedRepo)-2]
+		grepo        = splittedRepo[len(splittedRepo)-1]
 	)
 
+	for page <= maxPage && page > 0 {
+		rls, res, err := client.ListReleases(
+			ctx,
+			owner,
+			grepo,
+			&github.ListOptions{Page: page, PerPage: perPage},
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, r := range rls {
+			if release == nil {
+				release = r
+			} else if strings.Compare(*r.TagName, *release.TagName) == 1 {
+				release = r
+			}
+		}
+
+		page = res.NextPage
+	}
+
+	return release, nil
+}
+
+func GetLastVersion(repo string) (*version.Version, error) {
+	r, err := GetLatestRelease(context.Background(), repo)
+
 	if err != nil {
 		if e, ok := err.(*github.ErrorResponse); ok && e.Response.StatusCode == 404 {
 			return &version.Version{Version: *semver.MustParse("v0.0.0")}, nil
```

Comment:
with your implementation you never could get in here. you might need to ensure r == nil && err == nil